clang/test/Sema/debug-95854.cpp

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang_cc1 -std=c++23 -fsyntax-only -verify %s
```

Comment:
I think this may be a more appropriate place for the test:

https://github.com/llvm/llvm-project/blob/main/clang/test/SemaCXX/anonymous-union.cpp

We usually wrap tests from github issues in a namespace starting with `GH` and followed by the issue number for this case `namespace GH95854`

Reply:
Indeed.